### DISCUSSION TITLE
Can I Automatically Execute a Script Whenever Kitty Changes Themes?

### DISCUSSION THREAD

**Discussion by austinliuigi** [kovidgoyal/kitty#6014] [Q&A]
I would like to have the colors of my bash prompt be in sync with those of kitty's current theme. Is there a straightforward way to do this?

Thanks

**Comment by page-down** [comment]
There are two directions to think about, one is to proactively push change notifications to shells and the second is to let shells passively pull and check.

In the first approach, the shell needs to have a way to listen to external events in order to allow either foreground or background running processes to respond and execute code.
Otherwise you can only ask the shell running in the foreground to execute the code by sending escape sequences or keystrokes, while the shell in the background, e.g. running vim and then exiting, will not be updated.

And the time to send change notifications is when you ask kitty to change the theme, for example to reload the configuration or to change the colors one by one via remote control commands.
You can configure a shortcut to reload the prompt theme in the shell, and then monkey patch kitty to send that key to all windows that sitting at the shell prompt when reloading the configuration.

And the second approach, you can check whether the kitty configuration file has changed when drawing the next prompt, and then load the matching theme.

> Is there a straightforward way to do this?

Save the modification time of the kitty configuration file at shell startup, check the modification time when drawing shell prompt, and if it doesn't match, parse kitty configuration file and load the matching theme.
